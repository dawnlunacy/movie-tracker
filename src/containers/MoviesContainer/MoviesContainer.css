.scroll-wrapper {
    width: 100vh;
    height: 100vw;
    transform: rotate(-90deg) translateX(-100vh);
    transform-origin: top left;
    overflow-y: scroll;
    overflow-x: hidden;
    position: absolute;
    top: 500px;
}

.movies-container {
    display:flex;
    justify-content: center;
    flex-direction: row;
    margin-top: 50px;
    width: fit-content;
    transform: rotate(90deg) translateY(-105vh);
    transform-origin: top left;

}
