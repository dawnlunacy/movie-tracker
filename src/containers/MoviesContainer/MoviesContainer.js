import React from 'react';
import Movie from '../Movie/Movie';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import './MoviesContainer.css'

export const MoviesContainer = ({ movies, errorMessage, toggleFavorite, toggleStar }) => {
  const moviesToDisplay = movies.map(movie => {
    const { movie_id, title, poster_path, release_date, vote_average, overview } = movie;
    return <Movie
      key = {movie_id}
      movie_id = {movie_id}
      title = {title}
      poster_path = {poster_path}
      release_date = {release_date}
      vote_average = {vote_average}
      overview = {overview}
      toggleFavorite = {toggleFavorite}
      toggleStar = {toggleStar}
    />
  })

  return (
    <div className="scroll-wrapper">
      <main className="movies-container">
        {errorMessage && <p>{errorMessage}</p>}
        {moviesToDisplay}
      </main>
    </div>
  )
}

export const mapStateToProps = state => ({
  movies: state.movies,
  errorMessage: state.errorMessage
});

export default connect(mapStateToProps, null)(MoviesContainer);

MoviesContainer.propTypes = {
  movies: PropTypes.array.isRequired,
  errorMessage: PropTypes.string.isRequired,
  toggleFavorite: PropTypes.func.isRequired
}