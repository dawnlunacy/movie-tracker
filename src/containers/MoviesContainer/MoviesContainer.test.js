import React from 'react';
import { shallow } from 'enzyme';
import MoviesContainer from './MoviesContainer';

describe('MoviesContainer', () => {

  it('should match snapshot', () => {
    const wrapper = shallow(<MoviesContainer />)

    expect(wrapper).toMatchSnapshot();
  })

})
