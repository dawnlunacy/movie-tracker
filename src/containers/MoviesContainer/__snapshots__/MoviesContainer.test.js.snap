// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesContainer should match snapshot 1`] = `
<div
  className="scroll-wrapper"
>
  <main
    className="movies-container"
  >
    <Connect(Movie)
      key="999999"
      movie_id={999999}
      overview="During the 1980s, a failed stand-up comedian is driven insane and turns to a life of crime and chaos in Gotham City while becoming an infamous psychopathic crime figure."
      title="Joker"
    />
    <Connect(Movie)
      key="888888"
      movie_id={888888}
      overview="Maleficent and her goddaughter Aurora begin to question the complex family ties that bind them as they are pulled in different directions by impending nuptials, unexpected allies, and dark new forces at play."
      title="Maleficent: Mistress of Evil"
    />
  </main>
</div>
`;
